### PR TITLE
Style Book: Fix critical error when heading block is not registered

### PR DESCRIPTION
**Pull request by t-hamano**
Discovered this while reviewing #64038

## What?

This PR fixes a critical error in the Style Book screen when the heading block is not registered.

style-book

## Why?

This is because a non-existent block is being created via `createBlock()`.

## How?

Use the `getBlockType()` selector to pre-check if a heading block exists, and if not, return non-heading block examples.

## Testing Instructions

- Run the following code in your browser console: `wp.blocks.unregisterBlockType('core/heading')`
- Open the StyleBook.
- You won’t see the heading block, but the StyleBook should display fine.

### CODE REVIEW

File: packages/edit-site/src/components/style-book/index.js [hunk 1]
```diff
@@ -178,7 +145,31 @@ function getExamples() {
...
+		blocks: [ 1, 2, 3, 4, 5, 6 ].map( ( level ) => {
+			return createBlock( 'core/heading', {
+				content: sprintf(
+					// translators: %d: heading level e.g: "1", "2", "3"
+					__( 'Heading %d' ),
+					level
+				),
+				level,
+			} );
+		} ),
```
Comment (t-hamano) [inline packages/edit-site/src/components/style-book/index.js:169]: This is not directly related to this PR, but it is a code refactoring. The translator comment is the same as that of the `HeadingLevelDropdown` component:

https://github.com/WordPress/gutenberg/blob/27fbbd8e9b57495339a077ef0c8ba67a18392699/packages/block-editor/src/components/block-heading-level-dropdown/index.js#L55-L59

**Review by ramonjd — APPROVED** [review]
Good catch!

## Before

https://github.com/user-attachments/assets/dae1c840-3020-4374-a41f-0e10388c14c3


## After


https://github.com/user-attachments/assets/eed38488-1e17-4d3e-b492-a6276e2a9bf5
